src/datastore/postgres-notifier.ts

```diff
@@ -99,15 +109,27 @@ export class PgNotifier {
   }
 
   public async close() {
-    await this.subscriber.unlisten(this.pgChannelName);
-    await this.subscriber.close();
+    await this.subscriber?.unlisten(this.pgChannelName);
+    await this.subscriber?.close();
   }
 
   private async notify(notification: PgNotification) {
-    await this.subscriber
-      .notify(this.pgChannelName, { notification: notification })
-      .catch(error =>
-        logError(`Error sending PgNotifier notification of type: ${notification.type}`, error)
-      );
+    if (this.subscriber) {
+      await this.subscriber
+        .notify(this.pgChannelName, { notification: notification })
+        .catch(async error => {
+          if (error instanceof Error && error.message.includes('not queryable')) {
```

Comment:
What do you think of some of the reconnect and error related events https://github.com/andywer/pg-listen#instanceeventsonreconnect-listener-attempt-number--void ?